### DISCUSSION TITLE
Where store user Token in my Nuxt 3 app?

### DISCUSSION THREAD

**Discussion by AxelBriche** [nuxt/nuxt#10402] [Questions (Nuxt 2)]
Hi, I'm using Nuxt 3 with a fullstack approach, I'm using Pinia for the store, and Prisma as ORM to interact with my database.

Nuxt Auth is not compatible with Nuxt 3, I want to create a simple login/registration feature (with security in mind) and I would like to know where I can store the user's token in a safe place?

Cookies can be disabled in the browser, local storage is not available server-side.

Currently, after a successful login, I call my "security-store", generate a token, push it into the database, and store the token inside a cookie (client side). I plan to create a middleware that checks on each page call if the token is valid and if it matches the database user token (and refresh it?).

Is this a good practice and what do you recommend?
Thank you in advance for your answers.

**Comment by manh-go2joy** [comment]
I same issue

**Comment by novvac** [comment]
The user sends access data to the server (/login). The server checks the given data and searches the database for the corresponding user by returning a token in the form of a cookie with the flag httponly in the reply headers. The user sends a token for each request that retrieves sensitive data from the database (remember, always with an encrypted connection – https). If you are using axios, check this: https://stackoverflow.com/questions/43002444/make-axios-send-cookies-in-its-requests-automatically.

Set-Cookie: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

**Reply by chrisonderwater** [reply]
Little late to the party, but does this also work in the asyncData() calls? How can I make the token available in asyncData()?

**Comment by videmann** [comment]
Hi! On my project I use cookie. I'm setting it with /server/middleware/token.global.ts and use with my own useFetch wrapper, which has bearer authorization header

**Reply by ecaste** [reply]
Could you please provide some code for your server middleware ?
By the way, does this method work on page refresh ?
